packages/phishing-controller/src/PhishingDetector.test.ts

```diff
@@ -0,0 +1,979 @@
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type, name } = detector.check('blocked-by-first.com');
+
+          expect(result).toBe(true);
+          expect(type).toBe('blocklist');
+          expect(name).toBe('first');
+        },
+      );
+    });
+
+    it('blocks a domain when it is in the blocklist of the second config', async () => {
+      await withPhishingDetector(
+        [
+          {
+            allowlist: [],
+            blocklist: [],
+            fuzzylist: [],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+          {
+            allowlist: [],
+            blocklist: ['blocked-by-second.com'],
+            fuzzylist: [],
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type, name } = detector.check(
+            'blocked-by-second.com',
+          );
+
+          expect(result).toBe(true);
+          expect(type).toBe('blocklist');
+          expect(name).toBe('second');
+        },
+      );
+    });
+
+    it('prefers the first config when a domain is in both blocklists', async () => {
+      await withPhishingDetector(
+        [
+          {
+            allowlist: [],
+            blocklist: ['blocked-by-both.com'],
+            fuzzylist: [],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+          {
+            allowlist: [],
+            blocklist: ['blocked-by-both.com'],
+            fuzzylist: [],
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type, name } = detector.check('blocked-by-both.com');
+
+          expect(result).toBe(true);
+          expect(type).toBe('blocklist');
+          expect(name).toBe('first');
+        },
+      );
+    });
+
+    it('blocks a domain when it is in the fuzzylist of the first config', async () => {
+      await withPhishingDetector(
+        [
+          {
+            allowlist: [],
+            blocklist: [],
+            fuzzylist: ['fuzzy-first.com'],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+          {
+            allowlist: [],
+            blocklist: [],
+            fuzzylist: [],
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type, name } = detector.check('fuzzy-first.com');
+
+          expect(result).toBe(true);
+          expect(type).toBe('fuzzy');
+          expect(name).toBe('first');
+        },
+      );
+    });
+
+    it('blocks a "similar enough" domain when it is in the fuzzylist of the first config', async () => {
```

Comment:
Nit: Curious if this would describe the behavior better:

``` suggestion
    it('blocks a domain that is similar enough (within a tolerance) to a domain in the fuzzylist of the first config', async () => {